src/include/common/concurrent_bitmap.h

```diff
@@ -88,6 +88,106 @@ class RawConcurrentBitmap {
     return false;
   }
 
+  /**
+   * Looks for an unset bit in a T-sized word from the bitmap, starting at byte_pos.
+   * If an unset bit is found, returns true.
+   * Otherwise updates byte_pos to be the next place we should search.
+   * @tparam T signed fixed width integer type.
+   * @param[in,out] byte_pos invariant: next byte position we should search.
+   * @param[in,out] bits_left the number of valid bits remaining.
+   * @return true if an unset bit was found, false otherwise.
+   */
+  template <class T>
+  bool FindUnsetBit(uint32_t *byte_pos, uint32_t *bits_left) {
+    // for a signed integer, -1 represents that all the bits are set
+    if (reinterpret_cast<std::atomic<T> *>(&bits_[*byte_pos])[0].load() == -1) {
+      *byte_pos += static_cast<uint32_t>(sizeof(T));
+      // saturate bits_left at 0
```

Comment:
nit: "saturate" has unclear meaning here. maybe just say you want to prevent underflow